collector/processors/odigosconditionalattributes/README.md

```diff
@@ -0,0 +1,129 @@
+# **Odigos Conditional Attributes Processor**
+
+## Overview
+
+The **Odigos Conditional Attributes Processor** is a custom OpenTelemetry Collector processor that adds new attributes to spans based on the values of existing span attributes. It allows you to define rules to conditionally add static values, copy values from other attributes, or set a global default when no rules match.
+
+This processor is ideal for enriching spans with categorized or contextual information, improving observability.
+
+## Features
+
+- **Conditional Attribute Processing**: Add attributes based on specific span attribute values.
+- **Static Values**: Assign predefined values to new attributes.
+- **Dynamic Values**: Copy values from other span attributes.
+- **Global Default**: Set a fallback value when no rules apply or an attribute is missing.
+
+# Key Configuration Fields
+
+## Global Options
+
+| Field           | Description                                     | Required | Default   |
+|------------------|-------------------------------------------------|----------|-----------|
+| `global_default` | The value assigned when no rules match.         | Yes       | - |
+
+## Rule Options
+
+| Field               | Description                                                               | Required |
+|----------------------|---------------------------------------------------------------------------|----------|
+| `attribute_to_check` | The attribute in the span to evaluate.                                   | Yes      |
+| `values`             | A map of conditions to a list of actions for processing.                | Yes      |
+
+## Value Map Options
+
+Each key in the `values` map corresponds to a potential value of `attribute_to_check`. The associated value is a list of `Value` objects, each specifying how to process the span.
+
+| Field            | Description                                                                | Required |
+|-------------------|----------------------------------------------------------------------------|----------|
+| `new_attribute`  | The name of the new attribute to add to the span.                          | Yes      |
+| `value`          | A static string value to assign to the `new_attribute`.                   | No       |
+| `from_attribute` | The attribute whose value will be copied to the `new_attribute`.          | No       |
+
+# How It Works
+
+1. The processor checks the value of `attribute_to_check` for each span.
+2. It matches the value against the keys in `values`.
+3. For each match, it iterates through the list of `Value` objects:
+   - Assigns a static value (`value`) to the `new_attribute`.
+   - Copies a value from another attribute (`from_attribute`) to the `new_attribute`.
+4. If no match is found, assigns the `global_default` to all `new_attribute`s specified by the rule.
```

Comment:
If i understand correctly it will do it once at the time collector initialised
```
/Users/tamirdavid/workspace/odigos/collector/processors/odigosconditionalattributes/factory.go:31
```
`uniqueNewAttributes := calculateUniqueNewAttributes(cfg.(*Config))`